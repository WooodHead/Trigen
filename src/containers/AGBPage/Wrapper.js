import styled from 'styled-components';


export default styled.section`
  margin: 0 auto;
  width: 50%;
`;
