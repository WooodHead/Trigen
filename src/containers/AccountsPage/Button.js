/**
  * A restyled Button component
  */

import styled from 'styled-components';

import Button from 'components/Button';


export default styled(Button)`
  display: inline-block;
  margin: 5px;
`;
