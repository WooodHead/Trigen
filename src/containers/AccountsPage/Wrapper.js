import styled from 'styled-components';


export default styled.section`
  min-width: 50%;
  margin: 0 auto;
`;
