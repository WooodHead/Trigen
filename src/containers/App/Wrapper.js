/**
  * Styled Wrapper for the whole application
  *
  * @namespace App.Wrapper
  * @memberOf App
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  min-height: 100%;
  display: flex;
  flex-direction: column;
`;

export default Wrapper;
