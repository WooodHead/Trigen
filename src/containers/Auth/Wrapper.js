/**
  * Renders a styled div for the HomePage
  *
  * @namespace HomePage.Wrapper
  * @memberOf HomePage
  */

import styled from 'styled-components';


const Wrapper = styled.section`
  margin: 0 auto;
  width: 50%;
`;

export default Wrapper;
