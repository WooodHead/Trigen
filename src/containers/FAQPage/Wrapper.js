import styled from 'styled-components';


export default styled.section`
  width: 50%;
  margin: 0 auto;
`;

