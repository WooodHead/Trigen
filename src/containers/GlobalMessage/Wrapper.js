import styled from 'styled-components';


const Wrapper = styled.div`
  margin: 0 0 20px 0;
  padding: 10px 20px;

  border-radius: 4px;

  background: ${props => props.error ? props.theme.warningHover : props.theme.primaryHover};
  opacity: .6;
  color: #fff;
`;

export default Wrapper;
