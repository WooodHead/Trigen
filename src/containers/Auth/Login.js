import React, { PropTypes, Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import styled from 'styled-components';

import LoginForm from './LoginForm';
import Wrapper from './Wrapper';
import StyledLink from 'components/StyledLink';

import { loginSubmit } from './ducks/actions';


export class Login extends Component {
  handleSubmit = values => {
    this.props.loginSubmit(values);
  }

  render() {
    return (
      <Wrapper>
        <LoginForm onSubmit={this.handleSubmit} />
        <StyledLink to="/register" secondary>Register</StyledLink>
      </Wrapper>
    );
  }
}

Login.propTypes = {
  loginSubmit: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    loginSubmit,
  }, dispatch);
};

export default connect(null, mapDispatchToProps)(Login);
