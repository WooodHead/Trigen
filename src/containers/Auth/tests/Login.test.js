import React from 'react';
import { shallow } from 'enzyme';

import Button from 'components/Button';

import Wrapper from '../Wrapper';
import { Login } from '../Login';
import LoginForm from '../LoginForm';


const renderComponent = newProps => {
  const props = {
    loginSubmit: jest.fn(),
    ...newProps,
  };

  const renderedWrapper = shallow(<Login {...props} />);

  return {
    props,
    renderedWrapper,
  };
};

describe('<Login />', () => {
  it('should render <Wrapper />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(Wrapper).length).toMatchSnapshot();
  });

  it('should render <LoginForm />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(LoginForm).length).toMatchSnapshot();
  });

  it('should render <Button />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(Button).length).toMatchSnapshot();
  });

  it('should call loginSubmit() on handleSubmit', () => {
    const { renderedWrapper, props } = renderComponent();

    renderedWrapper.instance().handleSubmit('test');

    expect(props.loginSubmit).toHaveBeenCalledWith('test');
  });
});
