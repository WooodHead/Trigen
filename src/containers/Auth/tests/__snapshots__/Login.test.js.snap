exports[`<Login /> should render <Button /> 1`] = `1`;

exports[`<Login /> should render <LoginForm /> 1`] = `1`;

exports[`<Login /> should render <Wrapper /> 1`] = `1`;
